src/node/rpc/member_frontend.h

```diff
@@ -1268,13 +1270,15 @@ namespace ccf
         {
           node.initiate_private_recovery(args.tx);
         }
-        catch (const std::logic_error& e)
+        catch (const std::exception& e)
         {
           // For now, clear the submitted shares if combination fails.
+          auto error_msg =
+            fmt::format("Failed to initiate private recovery: {}", e.what());
+          LOG_FAIL_FMT(error_msg);
           share_manager.clear_submitted_recovery_shares(args.tx);
           args.rpc_ctx->set_response_status(HTTP_STATUS_INTERNAL_SERVER_ERROR);
-          args.rpc_ctx->set_response_body(
-            fmt::format("Failed to initiate private recovery: {}", e.what()));
+          args.rpc_ctx->set_response_body(std::move(error_msg));
```

Comment:
Since we are catching everything should be be sending it back to the user? this might leak confidential info